vms/platformvm/vm.go

```diff
@@ -133,9 +134,13 @@ func (vm *VM) Initialize(
 
 	rewards := reward.NewCalculator(vm.RewardConfig)
 
+	g, err := genesis.Parse(genesisBytes)
+	if err != nil {
+		return err
+	}
 	vm.state, err = state.New(
 		vm.db,
-		genesisBytes,
+		g,
```

Comment:
by pulling the byte parsing out of state and passing a well defined object instead, we open the way to simplify state testing. State is surprisingly not well tests (with a few notable exceptions), and part of the reason is the complex building of genesis state. This PR solve this problem